src/gateway/bridge/shard_messenger.rs

```diff
@@ -58,24 +58,17 @@ impl ShardMessenger {
     /// parameter:
     ///
     /// ```rust,no_run
-    /// # use tokio::sync::Mutex;
-    /// # use serenity::model::gateway::{GatewayIntents, ShardInfo};
-    /// # use serenity::model::id::ShardId;
     /// # use serenity::gateway::{ChunkGuildFilter, Shard};
-    /// # use std::sync::Arc;
-    /// #
-    /// # async fn run() -> Result<(), Box<dyn std::error::Error>> {
-    /// #     let mutex = Arc::new(Mutex::new("".to_string()));
-    /// #
-    /// #     let shard_info = ShardInfo {
-    /// #         id: ShardId(0),
-    /// #         total: 1,
-    /// #     };
-    /// #     let mut shard = Shard::new(mutex.clone(), "", shard_info, GatewayIntents::all(), None).await?;
-    /// #
+    /// # async fn run(mut shard: Shard) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
```suggestion
    /// # async fn run() -> Result<(), Box<dyn std::error::Error>> {
```
Hard to create an auto-applicable suggestion here, but the best way to do this would be to add a line as follows:
```rust
let mut shard: Shard = unimplemented!();
```
The other places you defined `async fn run(mut shard: Shard)` should also be changed as such.

Reply:
Why? A lot of other examples take Client as an argument and I don't see the benefit at all of doing `unimplemented!()`.

Reply:
yeah, you're right - this can stay, it was just a minor nit